src/lib.rs

```diff
@@ -108,38 +128,50 @@ main!(|_cli: Cli| {
         style("...").bold()
     );
 
-    let dir_name = if force { name.raw() } else { name.kebab_case() };
-    let project_dir = env::current_dir()
-        .unwrap_or_else(|_e| ".".into())
-        .join(dir_name);
-
-    ensure!(
-        !project_dir.exists(),
-        "Target directory `{}` already exists, aborting.",
-        project_dir.display()
-    );
-
-    git::create(&project_dir, config)?;
+    if project_dir.exists() {
+        None
+    } else {
+        Some(project_dir)
+    }
+}
 
-    let template = template::substitute(&name, force)?;
+fn progress(name: &ProjectName, dir: &PathBuf, force: bool) {
+    let template =
+        template::substitute(name, force).expect("Error: Can't substitute the given name.");
```

Comment:
i think we'd like to still use the `?` mark operator- is there a reason you replaced it with `expect`?